### ISSUE TITLE
Exiting KOReader on Kobo Libra H2O does not trigger indexing of the books in Kobo Reader

### ISSUE THREAD

**Issue by maichai** [koreader/koreader#7375]
* KOReader version: 2021.02
* Device: Kobo Libra H2O

#### Issue

epubs downloaded through OPDS are visible in KOReader and Plato, but not in the built-on Kobo Reader (FW 4.26.16704)

#### Steps to reproduce

- start KOReader
- download epub using OPDS to /mnt/onboard/books
- exit KOReader (reboot device)
- on Libra H20: epub is not indexed and not visible
- on Aura (same FW version): epub is indexed and is visible

- connect Libra to USB in the built-in Kobo Reader USB storage mode (NOT in the KOReader USB storage mode!)
- disconnect Libra from USB
- epubs now are indexed and visible also on the Libra H2O

It seems that exiting KOReader on the Aura does trigger a re-indexing of the epubs, whereas exiting KOReader on the Libra H2O does not trigger a re-indexing. However, a re.indexing seems to happen when exiting the USB storage mode of the built-in Kobo Reader.

**Comment by NiLuJe** [comment]
xref and canonical answer: https://www.mobileread.com/forums/showpost.php?p=4098526&postcount=7

Put into KOReader's context
* Nickel is killed when KOReader starts.
* Nickel doesn't index on startup.

TL;DR: Use NM's `rescan_books` when you're back in Nickel.

**Comment by NiLuJe** [comment]
As for why it appears to work on the Aura: it doesn't, actually, not for the internal storage ;).

But, for the sd card, an eject/insert event is faked on startup, and *that* triggers a re-indexing of the sd card itself.

**Comment by NiLuJe** [comment]
Actually, we no longer really do a complete fake eject/insert event, because that led to Nickel whinging about it on current FW versions.

What we do is simply unmount it before Nickel starts, and then Nickel & and udev trigger take care of the rest.

This obviously only happens if there's an SD card physically inserted in the device, hence the potential for slightly different behaviors.

**Comment by maichai** [comment]
Perfect explanation. What I did not say but it totally fits into the picture: on the Aura I am downloading the books to /mnt/sd/book instead of /mnt/onboard/book.

**Comment by NiLuJe** [comment]
Yup, that was my assumption given what you detailed ;).
